### DISCUSSION TITLE
configuring-environment-specific-services-in-asp-net-core/

### DISCUSSION THREAD

**Discussion by andrewlock** [andrewlock/blog-comments#162] [Announcements]
# configuring-environment-specific-services-in-asp-net-core/

In this post I show how you can configure dependency injection so that different services will be injected when you are in a development environment.

https://andrewlock.net/configuring-environment-specific-services-in-asp-net-core/

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at September 16 2016, 09:44, in reply to [Muhammad Rehan Saeed](https://disqus.com/by/RehanSaeedUK/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/162#discussioncomment-6005363)</em>

---

<p>Great point, I had forgotten about those, definitely makes it cleaner:)</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at September 16 2016, 09:45, in reply to [Spetsen](https://disqus.com/by/spetsen/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/162#discussioncomment-6005365)</em>

---

<p>Yep, see my next post! :) <a href="https://andrewlock.net/configuring-environment-specific-services-in-asp-net-core-part-2/" rel="nofollow noopener" title="https://andrewlock.net/configuring-environment-specific-services-in-asp-net-core-part-2/">Configuring environment specific services in ASP.NET Core - Part 2</a></p>

[a comment by an automated account is not shown]

**Comment by paroze** [comment]
Hello Andrew,

How to achieve this in .net 6 minimal api?

Thx

**Reply by andrewlock** [reply]
It's essentially the same, for example:

```csharp
var builder = WebApplication.CreateBuilder(args);

if (builder.Environment.IsDevelopment())
{
    builder.Services.AddTransient<ICurrencyRateService, DummyCurrencyRateService>();
}
else
{
    builder.Services.AddTransient<ICurrencyRateService, ExternalCurrencyRateService>();
}

var app = builder.Build();
```

**Reply by paroze** [reply]
Thx :)
